### ISSUE TITLE
As a Dfinity employee, I can use @dfinity/identity-provider w/ Pim's Design Phase 1, so I can gather stakeholder/end-user feedback

### ISSUE THREAD

**Issue by gobengo** [dfinity/agent-js#110]
## Goal:
* Update the identity-provider UI that @codelemur came up with to match the simplest-possible version of @pimentel-angelica's design artifact.
* Unblock design research or other end-user acceptance testing of the simplest workflow without implementing all the user stories.
* Inform prioritization for what to do immediately after (e.g. Identity backup/restore)

## Demo
1. Visit https://identity-provider.sdk-test.dfinity.network/relying-party-demo/
2. Follow the primary calls-to-action. It should look generally like the wireframe-fidelity design below.
3. You should end up back at the relying party along with enough authentication data to unblock making requests against IC 

## Design
![Slice 1](https://user-images.githubusercontent.com/171782/101548626-06957080-3961-11eb-9568-b17686366051.png)
[click here for png/svg in slack](https://dfinity.slack.com/archives/C01BJ73A893/p1607465932460500)

## Recommended Followup

* Implement any variations in UX that should happen when going through this authentication flow other than the first time, i.e. after the root identity creation,. i.e. "re-authentication for same-device, same-canister"
* Add functionality to support backing up the root identity that the IDP guides the end-user through creating
* Add functionality to support importing a root identity instead of the IDP creating one for the end-user

## Acceptance Criteria

Check all these after the tasks/progress appear to be complete.

* [x] On any of these multiple screens, the end-user should be able to refresh the page and continue their flow without starting over (i.e. give each screen in the flow a distinct URL)
* [x] This works well on Chrome for Mac
* ~This works well on Safari for Mac~
  * Safari not great right now. Fix in followup. May require specific options to support FaceID https://webkit.org/blog/11312/meet-face-id-and-touch-id-for-the-web/
* [x] This works well on Firefox for Mac
* ~This works well on Safari for iOS~
  * Safari iOS not great right now. Fix in followup. May require specific options to support FaceID https://webkit.org/blog/11312/meet-face-id-and-touch-id-for-the-web/

## Tasks / Progress

* [x] Implement Screen 3: "Onboarding"
* [x] WebAuthn: https://github.com/dfinity/agent-js/issues/144
* [x] Implement Screen 5: "Profile Account" - i.e. show the user the identity we've chosen/created for them
* Implement Screen 6: "Consent/Canister" - i.e. end-user authorizes Canister to use their identity
  * [x] initial good design
  * [x] Document exactly how an ic-idp Relying Party (e.g. a canister / auth lib) should provide canister IDs as the 'targets'[] that scope delegation to only a couple targets. (card: https://github.com/orgs/dfinity/projects/6#card-52318854)
  * https://github.com/dfinity/agent-js/pull/136
  * [x] After consent, the resulting AuthenticationResponse should actually have its sender_delegation scoped to those targets[]
* [x] Implement Screen 7: "IC Entry/Successful" - i.e. let the end-user know authn is done, and let them click to be returned to the RP Canister

## Questions

* @gobengo 
  * [x] What should happen if the user declines to consent to the WebAuthn Invocation?
  * [x] What happens if the user clicks 'Not Now' on 'Consent/Canister'?
    * assumption/suggestion: If the user really wants to bail out, all we can do is redirect back to the RP Canister with a useful error code that lets the canister show a message like: "Looks like you decided not to complete authentication. Click here if you want to try again"

Unblocks:
* https://github.com/dfinity/agent-js/issues/143
*   Update webauthn_tester RP to add its canister principals IDs to AuthenticationRequest.scope. Otherwise it might not work? Test after this.
